cmd/node/dialback.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/blocklessnetwork/b7s/models/blockless"
+)
+
+func purgeDialbackPeers(store blockless.PeerStore) error {
+
+	// NOTE: If the peer count grows too much in the future - we should adopt a more iterative approach.
```

Comment:
We should cap this at 100, as that is the limit of the DHT in the PubSub Mesh. Unless we think there's also a reason to retain direct dials.

Reply:
It's opaque, but we actually do limit this to a 100 in the `host` package - [config link](https://github.com/blocklessnetwork/b7s/blob/main/host/config.go#L15) and [usage link](https://github.com/blocklessnetwork/b7s/blob/main/host/dht.go#L121).

Taking this into account, this should definitely be part of consideration when retrieving dialback peers as anything over a hundred is a waste. I was planning on working on peers in a bit more detail, but I don't have a good plan yet. I was thinking it would be nice to have a reliable timestamp when the peer was last seen. Currently we only know time of last connection - but one could connect once and remain connected for a long time. (Perhaps the node periodically enumerates peers that are connected and updates the DB?)

If we had this info, we could essentially do `select * from peers order by last_seen desc`, and get the freshest peers first. Then if we trim the list to X we wouldn't lose anything.